src/package_source.rs

```diff
@@ -79,7 +85,7 @@ impl PackageSourceMap {
     }
 }
 
-pub trait PackageSource {
+pub trait PackageSourceImpl: Sync + Send + Clone {
```

Comment:
The trait name here is a bit awkward since it makes for `impl PackageSourceImpl` in other places, maybe something like `PackageSourceProvider` could work better?

Reply:
Yeah I wasn't sure what to call this so this was mostly just a temporary name to see if this worked. I like `PackageSourceProvider` more!